vm.go

```diff
@@ -306,12 +306,29 @@ type vm struct {
 	interrupted   uint32
 	interruptVal  interface{}
 	interruptLock sync.Mutex
+
+	lastFunctionTicks uint64
 }
 
 type instruction interface {
 	exec(*vm)
 }
 
+func (vm *vm) setPrg(prg *Program) {
+	if !vm.r.disableTickMetrics {
+		currentTicks := vm.r.Ticks()
+		if vm.prg != nil {
+			if vm.r.functionTickMetrics == nil {
+				vm.r.functionTickMetrics = make(map[string]uint64)
+			}
+			functionName := vm.prg.src.Name() + "_" + string(vm.prg.funcName)
```

Comment:
I originally had an optimization to save this value on the prg object, but that was when this was being hit for every tick. I don't think that optimization is necessary anymore.